iso15118/shared/comm_session.py

```diff
@@ -475,6 +475,10 @@ async def rcv_loop(self, timeout: float):
                 return
 
             try:
+                if hasattr(self.comm_session, "evse_controller"):
+                    await self.comm_session.evse_controller.set_present_protocol_state(
+                        str(self.current_state)
+                    )
                 # This will create the values needed for the next state, such as
                 # next_state, next_v2gtp_message, next_message_payload_type etc.
                 await self.process_message(message)
```

Comment:
Looks like it might suffice to call set_present_protocol_state once after the above self.process_message?
```
                if hasattr(self.comm_session, "evse_controller"):
                    await self.comm_session.evse_controller.set_present_protocol_state(
                        # noqa
                        str(self.current_state)
                    )
```

Reply:
Done